### ISSUE TITLE
Delete segments of a given time range

### ISSUE THREAD

**Issue by yupeng9** [apache/pinot#8114]
There are occasional asks from users to delete segments of a given range. (.e.g from 4days ago to yesterday). Since the segment metadata has the range info, it's possible for such deletion at the day's level for offline tables.

**Comment by walterddr** [comment]
I think we need a more formal definition for this feature before we consider the implementation. 

For example, 
- do we only support absolutely aligned segment time boundaries? (e.g. do not touch any segment that are across the start/end timestamp)
  - if so, what happen to those that cross the start/end timestamp?
  - also what happen if data were reuploaded?

also i felt like this can be achieve via a 2 step action - first query the segments by time range, and then use the delete segment endpoint.

**Comment by yupeng9** [comment]
Agreed. Perhaps there's a way for the user to review the identified segments (via dryrun), and then choose.
